### PR TITLE
Periodically flush large in-progress transactions

### PR DESCRIPTION
**Pull request by nbiscaro**
## Summary

Allows active streaming transactions to incrementally flush data to disk before the transaction fully commits. This improves memory management for large or long-running streams.

Rows from an ongoing streaming transaction can now be periodically flushed to disk slices stored in the `transaction_stream_state`. On commit, we update the `writer_lsn` for each of the disk slices to the commit LSN. We then append these disk slices to the `next_snapshot_task` to be processed on the next snapshot.

In the case of stream abort, we simply drop the transaction state and never add the disk slices to the main snapshot.

Tests the following edge cases: 

- streamed transaction deletes from main mem slice
- streamed transaction deletes from streamed mem slice
- streamed transaction deletes from flushed row within same transaction
- streamed transaction deletes from flush row from previous transaction
- streamed transaction flushes uncommitted rows then aborts

## Related Issues

Closes https://github.com/Mooncake-Labs/moonlink/issues/132

TODO: Clean up dead files that were periodically flushed from aborted transaction. https://github.com/Mooncake-Labs/moonlink/issues/133

## Changes

- 
- 
- 

## Checklist

- [ ] Code builds correctly
- [ ] Tests have been added or updated
- [ ] Documentation updated if necessary
- [ ] I have reviewed my own changes

### CODE REVIEW

File: src/moonlink/src/table_handler.rs [hunk 1]
```diff
@@ -74,7 +76,16 @@ impl TableHandler {
                     match event {
                         TableEvent::Append { row, xact_id } => {
                             let result = match xact_id {
-                                Some(xact_id) => table.append_in_stream_batch(row, xact_id),
+                                Some(xact_id) => {
+                                    let res = table.append_in_stream_batch(row, xact_id);
+                                    if table.should_transaction_flush(xact_id) {
```
[the remark this answers is not in the document]
Comment (nbiscaro) [reply src/moonlink/src/table_handler.rs:81]: Nice catch! There is a follow up issue for this here https://github.com/Mooncake-Labs/moonlink/issues/68

In the meantime we can do something like this https://github.com/Mooncake-Labs/moonlink/issues/132

File: src/moonlink/src/storage/mooncake_table.rs [hunk 2]
```diff
@@ -352,7 +365,39 @@ impl MooncakeTable {
...
+            metadata.path.clone(),
+            batches,
+            None,
+            index,
+        );
+        // TODO(nbiscaro): Find longer term solution that allows aysnc write
+        disk_slice.write()?;
+        Ok(disk_slice)
+    }
+
```
[the remark this answers is not in the document]
Comment (nbiscaro) [reply src/moonlink/src/storage/mooncake_table.rs:387]: nah
